.github/workflows/build-dotnet.yml

```diff
@@ -10,11 +10,12 @@ on:
 jobs:
   build-and-test-dotnet:
     name: Build, Test
-    runs-on: ${{ matrix.os-artifact[0] }}
+    # runs-on: ${{ matrix.os-artifact[0] }}
+    runs-on: macos-latest
     strategy:
       fail-fast: false
-      matrix:
-        os-artifact: [ [ubuntu-latest, linux], [windows-latest, windows], [macos-latest, macos] ]
+      # matrix:
+      #   os-artifact: [ [ubuntu-latest, linux], [windows-latest, windows], [macos-latest, macos] ]
```

Comment:
we should be testing on multiple platforms.

Reply:
Also, forcing everything to macos is slower and more expensive on GHA.

Reply:
We can still test on all platforms, but not with this method. We can create and build the package on MacOS, and feed it to the tests and run them on all platforms. We don't need to test the compilation itself on all platforms.

Reply:
My concern with this is that it makes macos the only platform where development and CI/CD work can be done. That itself raises vendor lock-in issues.

Reply:
You can definitely do development on other platforms, either by building for net6.0 only or just removing the `-ios` target framework.